lib/tasks/upgrade.rake

```diff
@@ -0,0 +1,48 @@
+# TRANSLATORS: do not translate
+desc <<-END_DESC
+  This task runs predefined upgrade steps.
+
+  Examples:
+    rake upgrade:run
+    rake upgrade:mark_as_ran TASK_NAME=some_rake_task,another_rake_task
+
+END_DESC
+
+namespace :upgrade do
+  task :run => :environment do
+    ENV['FOREMAN_UPGRADE'] = '1'
+
+    total = UpgradeTask.needing_run.count
+    UpgradeTask.needing_run.each_with_index do |task, index|
+      count = "#{index + 1}/#{total}"
+
+      message = if task.long_running?
+                  _("Upgrade Step %{count}: %{name}, This may take a long while.")
+                else
+                  _("Upgrade Step %{count}: %{name}")
+                end
+
+      puts '============================================='
+      puts message % {:name => task.name, :count => count}
+      task.mark_as_ran! if run_task(task)
+    end
+  end
+
+  def run_task(task)
+    begin
```

Comment:
You know, this is very similar to what we were discussing a couple days ago. IMO, this should use the simplified form:

```rb
def run_task(task)
  Rake::Task[task.task_name].execute
  true
rescue => e
  ...
end
```

I'm not saying you should change it, just that there's a better way ;)